src/generate.ts

```diff
@@ -304,38 +345,129 @@ class Generator {
   }
 }
 
+export interface GenerateOptions {
+  types: boolean
+  request: boolean
+  operations: boolean
+  prettier: boolean
+  withDoc: boolean
+  patchScript?: string
+  onParsed?: (parsed: ParsedOpenApi) => void
+}
+
+const componentTypes = [
+  "schemas",
+  "responses",
+  "examples",
+  "requestBodies",
+  "headers",
+  "securitySchemes",
+  "links",
+  "callbacks"
+]
+
 export async function generate(
-  docOrPathToDoc: string | OpenAPIV3.Document
+  docPath: string | string[],
+  generateOptions: Partial<GenerateOptions> = {}
 ): Promise<string> {
-  const doc = (await bundle(docOrPathToDoc)) as OpenAPIV3.Document
-
-  const generator = new Generator(doc)
+  let doc: OpenAPIV3.Document
+  if (!Array.isArray(docPath)) {
+    docPath = [docPath]
+  }
+  for (const path of docPath) {
+    const newDoc = (await SwaggerParser.bundle(path)) as OpenAPIV3.Document
+    if (!doc) {
+      doc = newDoc
+      if (!doc.components) {
+        doc.components = {}
+      }
+    } else {
+      // merge paths, do not override
+      for (const [path, pathItemObj] of Object.entries(newDoc.paths)) {
+        if (!doc.paths[path]) {
+          doc.paths[path] = pathItemObj
+        }
+      }
+      // merge types
+      if (newDoc.components) {
+        for (const componentType of componentTypes) {
+          if (newDoc.components[componentType]) {
+            for (const [key, val] of Object.entries(
+              newDoc.components[componentType]
+            )) {
+              if (!doc.components[componentType]) {
+                doc.components[componentType] = {}
+              }
+              if (!doc.components[componentType][key]) {
+                doc.components[componentType][key] = val
+              }
+            }
+          }
+        }
+      }
+    }
+    if (generateOptions.patchScript) {
+      const patchScript = require(generateOptions.patchScript)
+      doc = await patchScript.patch(doc, SwaggerParser)
+    }
+  }
+  const options: GenerateOptions = {
+    types: true,
+    request: true,
+    operations: true,
+    prettier: true,
+    withDoc: true,
+    patchScript: null,
+    ...generateOptions
+  }
+  const generator = new Generator(doc, options.withDoc)
+  if (options.onParsed) {
+    options.onParsed(generator)
+  }
 
-  let messyOutput =
-    "// This file is generated by [oats][0] and should not be edited by hand.\n//\n// [0]: https://github.com/influxdata/oats\n\n"
+  let output = ""
```

Comment:
what makes this output not messy?

Reply:
I changed the code to add several flags that are required by influxdb-client-js. After my changes, the `messyOutput` variable was used in 10 places, so I simplified the code to just `output`. There is still just one output that can be formatted by prettier.